ts/store/middlewares/navigationHistory.ts

```diff
@@ -10,13 +13,30 @@ import {
 import { Action, Dispatch, MiddlewareAPI } from "../actions/types";
 import { GlobalState } from "../reducers/types";
 
+/**
+ * if the user presses back button twice in a time window of exitConfirmThreshold milliseconds
+ * then app will be closed
+ */
+const exitConfirmThreshold = 2000 as Millisecond;
+
+/**
+ * exitApp will be called if user press twice back when
+ * navigation history is empty. Remember BackHandler.exitApp() works
+ * on Android platforms (and others) but not on iOS platforms
+ */
+const exitApp = () => BackHandler.exitApp();
+
 // tslint:disable-next-line:cognitive-complexity
 export function createNavigationHistoryMiddleware(): Middleware<
   Dispatch,
   GlobalState
 > {
   // tslint:disable-next-line:no-let
   let firstRun = true;
+  // tslint:disable-next-line:no-let
+  let exitRequestTime: Millisecond = 0 as Millisecond;
+  // tslint:disable-next-line:no-let
+  let isExitRequest = false;
```

Comment:
You don't need another var. Use
lastExitRequestTime: Option<Millisecond>

and reset it to "none" when no more valid.